x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTaskIndexService.java

```diff
@@ -307,6 +325,35 @@ public void getResponse(AsyncExecutionId asyncExecutionId,
         ));
     }
 
+
+    /**
+     * Gets the status response of the async search from the index.
+     * This should be called after unsuccessful attempt to retrieve a status from the Task Manager.
+     * As the corresponding task doest' exist, we assume that the async search request
+     * has been completed, and for a status response we are only interested in its expiration time.
+     * @param asyncExecutionId – id of the async search
+     * @param completedStatusProducer – a producer of the status response of the completed async task,
+     *   where necessary fields are only id and expiration time.
+     * @param listener – listener to report result to
+     */
+    public void getStatusResponse(
+        AsyncExecutionId asyncExecutionId, BiFunction<String, Long, R> completedStatusProducer, ActionListener<R> listener) {
```

Comment:
Can we reuse `getEncodedResponse` ? We'd need to disable the authentication check but that can be an option on the function directly ?

Reply:
Similarly to a previous comment, I opted out to have a separate method for this .  Please let me know if we still want to have a single method.